### ISSUE TITLE
Docs: no-mixed-operators - allowSamePrecedence option docs improvement

### ISSUE THREAD

**Issue by munkychop** [eslint/eslint#9962]
**Tell us about your environment**
N/A (macOS Sierra 10.12.6)

* **ESLint Version:**
v4.17.0

* **Node Version:**
N/A (v8.9.1)

* **npm Version:**
N/A (v5.5.1)

**What parser (default, Babel-ESLint, etc.) are you using?**
N/A (default)

**Please show your full configuration:**
N/A

**What did you expect to happen?**
The `allowSamePrecedence` option of the `no-mixed-operators` rule should correctly document "Examples of incorrect code for this rule with `{"allowSamePrecedence": false}` option".

**What actually happened? Please include the actual, raw output from ESLint.**
The documentation is incorrect, as it is the same as "Examples of correct code for this rule with {"allowSamePrecedence": true} option":
```js
/*eslint no-mixed-operators: ["error", {"allowSamePrecedence": true}]*/

// + and - have the same precedence.
var foo = a + b - c;
```
```js
/*eslint no-mixed-operators: ["error", {"allowSamePrecedence": false}]*/

// + and - have the same precedence.
var foo = a + b - c;
```

This should simply be changed to something like this:
```js
/*eslint no-mixed-operators: ["error", {"allowSamePrecedence": false}]*/

// + and * have different precedence.
var foo = a + b * c;
```

**Comment by platinumazure** [comment]
I'm not sure I understand.

The purpose of the rule is to require parentheses around an expression with two or more consecutive binary operators in it. The `allowSamePrecedence` option relaxes this requirement when the two operators are equal precedence.

So if `allowSamePrecedence` is false, then `var foo = a + b - c;` is in fact incorrect code: The rule is flagging *all* mixed/multiple binary operator expressions, even if they are the same precedence. That same example is correct (i.e., no lint warning) if `allowSamePrecedence` is true, because addition and subtraction have the same precedence.

The example `var foo = a + b * c;` should be incorrect whether `allowSamePrecedence` is true or false, because addition and multiplication are not the same precedence in any case. So this rule would then require the user to specify `var foo = a + (b * c);` (assuming the user knew what they were doing with the operators in the first place).

Am I missing something?

**Comment by munkychop** [comment]
Ah I see I've slightly misread the docs. I think I was confused due to there being no 'correct' example for `allowSamePrecedence:false`, although, I suppose code such as `var foo = a + (b - c);` is undesirable anyway, so this is really a non-issue.

Thanks for clearing things up.

**Comment by platinumazure** [comment]
@munkychop I think the natural/correct example in that case would be `var foo = (a + b) - c;`. Would you like to submit a PR?

**Comment by munkychop** [comment]
@platinumazure sure, I can do that. Maybe it would help clarify things for others :)

**Comment by munkychop** [comment]
I've created the PR. Let me know if there's anything more to add to that :)
